zerver/lib/topic.py

```diff
@@ -181,6 +182,15 @@ def update_messages_for_topic_edit(
         for m in messages_list:
             m.set_topic_name(topic_name)
 
+    if is_content_edited:
+        content_edit_event_keys = [
+            "prev_content",
+            "prev_rendered_content",
+            "prev_rendered_content_version",
+        ]
+        for key in content_edit_event_keys:
+            del edit_history_event[key]
+
```

Comment:
I think from an abstraction boundaries perspective, I'd prefer for us to just pass a new `edit_history_event` for the propagated messages constructed in the caller, rather than adding this new parameter.

Reply:
Okay, updated the PR.